lib/Parser/RegexRuntime.cpp

```diff
@@ -1180,270 +1187,73 @@ namespace UnifiedRegex
+    int BaseName##n##Inst::Print(DebugWriter* w, Label label, const Char* litbuf) const         \
+    {                                                                                           \
+        PRINT_RE_BYTECODE_BEGIN("SwitchAndConsume"#n);                                          \
+        PRINT_MIXIN(SwitchMixin<n>);                                                            \
+        PRINT_RE_BYTECODE_MID();                                                                \
+        PRINT_BYTES(SwitchMixin<n>);                                                            \
+        PRINT_RE_BYTECODE_END();                                                                \
     }
-#endif
-
-    // ----------------------------------------------------------------------
-    // SwitchAndConsume20Inst (optimized instruction)
-    // ----------------------------------------------------------------------
-
-    inline bool SwitchAndConsume20Inst::Exec(REGEX_INST_EXEC_PARAMETERS) const
-    {
-        if (inputOffset >= inputLength)
-            return matcher.Fail(FAIL_PARAMETERS);
-#if 0
-        int l = 0;
-        int h = numCases - 1;
-        while (l <= h)
-        {
-            int m = (l + h) / 2;
-#if ENABLE_REGEX_CONFIG_OPTIONS
-            matcher.CompStats();
-#endif
-            if (cases[m].c == input[inputOffset])
-            {
-                inputOffset++;
-                instPointer = matcher.LabelToInstPointer(cases[m].targetLabel);
-                return false;
-            }
-            else if (cases[m].c < input[inputOffset])
-            {
-                l = m + 1;
-            }
-            else
-            {
-                h = m - 1;
-            }
-        }
 #else
-        const int localNumCases = numCases;
-        for (int i = 0; i < localNumCases; i++)
-        {
-#if ENABLE_REGEX_CONFIG_OPTIONS
-            matcher.CompStats();
-#endif
-            if (cases[i].c == input[inputOffset])
-            {
-                inputOffset++;
-                instPointer = matcher.LabelToInstPointer(cases[i].targetLabel);
-                return false;
-            }
-            else if (cases[i].c > input[inputOffset])
-            {
-                break;
-            }
-        }
-#endif
-
-        instPointer += sizeof(*this);
-        return false;
-    }
-
-#if ENABLE_REGEX_CONFIG_OPTIONS
-    int SwitchAndConsume20Inst::Print(DebugWriter* w, Label label, const Char* litbuf) const
-    {
-        PRINT_RE_BYTECODE_BEGIN("SwitchAndConsume20");
-        PRINT_MIXIN(SwitchMixin<20>);
-        PRINT_RE_BYTECODE_MID();
-        PRINT_BYTES(SwitchMixin<20>);
-        PRINT_RE_BYTECODE_END();
-    }
-#endif
+#define COMP_STATS
+#define SwitchAndConsumeInstPrintImpl(BaseName, n)
+#endif
+
+#define SwitchAndConsumeInstImpl(BaseName, n) \
+    inline bool BaseName##n##Inst::Exec(REGEX_INST_EXEC_PARAMETERS) const                       \
```

Comment:
Why preprocessor rather than template?

Reply:
Arbitrary... Template would be better, but requires some other shenanigans that have made my other refactorings take longer. I wanted to get eyes on this change soon. I'll see how hard it would be to refactor to templates here.